pyp2rpm/name_convertor.py

```diff
@@ -200,3 +202,13 @@ def rpm_name(self, name, python_version=None):
 
 def canonical_form(name):
     return name.lower().replace('-', '').replace('_', '')
+
+
+class AutoProvidesNameConvertor(NameConvertor):
+    """Name convertor based on Automatic Provides for Python RPM Packages."""
+
+    non_alphanumeric_regexp = re.compile(r'[^0-9a-zA-Z]+')
+
+    def rpm_name(self, name, python_version=settings.DEFAULT_PYTHON_VERSION):
+        canonical_name = self.non_alphanumeric_regexp.sub('-', name).lower()
```

Comment:
Please use setuptool's [safe_name](https://github.com/pypa/setuptools/blob/master/pkg_resources/__init__.py#L1376) function, which is the one [used](https://github.com/rpm-software-management/rpm/blob/master/scripts/pythondistdeps.py#L136) to add an automatic provide for the package (if you trace down `dist.key`).
You can import it as: `from pkg_resources import safe_name`
Basically the only difference is that the dot is also allowed in the canonical name. And for example `zope.testing` should not be converted to `zope-testing`. Please also add this example to `test_rpm_name` parameterization.